src/builtins/core/datetime.rs

```diff
@@ -682,8 +683,33 @@ impl PlainDateTime {
         Ok(Self::new_unchecked(result, self.calendar.clone()))
     }
 
+    pub fn to_zoned_date_time_with_provider(
+        &self,
+        time_zone: &TimeZone,
+        disambiguation: Disambiguation,
+        provider: &impl TimeZoneProvider,
+    ) -> TemporalResult<ZonedDateTime> {
+        // 6. Let epochNs be ? GetEpochNanosecondsFor(timeZone, dateTime.[[ISODateTime]], disambiguation).
+        let epoch_ns = time_zone.get_epoch_nanoseconds_for(self.iso, disambiguation, provider)?;
+        // 7. Return ! CreateTemporalZonedDateTime(epochNs, timeZone, dateTime.[[Calendar]]).
+        Ok(ZonedDateTime::new_unchecked(
+            Instant::from(epoch_ns),
+            self.calendar.clone(),
+            time_zone.clone(),
+        ))
+    }
+
+    pub fn to_plain_date(&self) -> TemporalResult<PlainDate> {
```

Comment:
nit: add a temporal_capi implementation of `to_plain_date` for the FFI

This one may be a little harder because it's dealing with the diplomat bindings, but maybe you can take a crack at it. Changes should be made [here](https://github.com/boa-dev/temporal/blob/main/temporal_capi/src/plain_date_time.rs#L251).